### ISSUE TITLE
Edge poor performance

### ISSUE THREAD

**Issue by BryceAG** [phetsims/molecules-and-light#175]
Test device: Hanson

Operating System: Win 10

Browser: Edge

Problem description: performance is slow. Particles at full speed reduce frame rate to 8, and at slow speeds reduce it to 20. Even if no particles are moving (paused) trying to drag the light source slider is delayed, and highlights for which molecule the mouse hovers over is also delayed. Selecting a molecule or play/pause is also delayed

Steps to reproduce: load sim on Edge

Severity:

Screenshots:
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/15327201/35301853-f5e390da-0049-11e8-96d7-27fd3d2a97b9.gif)

Troubleshooting information (do not edit):
Name: ‪Molecules and Light‬
URL: https://www.colorado.edu/physics/phet/dev/html/molecules-and-light/1.4.0-rc.1/molecules-and-light_en.html?profiler
Version: 1.4.0-rc.1 2018-01-15 23:20:04 UTC
Features missing: touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.36 Edge/16.16299
Language: en-US
Window: 1280x603
Pixel Ratio: 2.5/1
WebGL: WebGL 1.0
GLSL: WebGL GLSL ES 1.0
Vendor: Microsoft (Microsoft Edge)
Vertex: attribs: 16 varying: 15 uniform: 512
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 16384x16384
OES_texture_float: true
Dependencies JSON: {"assert":{"sha":"928741cf","branch":"HEAD"},"axon":{"sha":"104f3dcf","branch":"HEAD"},"babel":{"sha":"32d7db04","branch":"master"},"brand":{"sha":"53cf227b","branch":"HEAD"},"chipper":{"sha":"6f8b4c13","branch":"HEAD"},"dot":{"sha":"ee4d0e25","branch":"HEAD"},"joist":{"sha":"1188c469","branch":"HEAD"},"kite":{"sha":"e46f990e","branch":"HEAD"},"molecules-and-light":{"sha":"e22bca76","branch":"HEAD"},"nitroglycerin":{"sha":"ccaab9cd","branch":"HEAD"},"phet-core":{"sha":"88a51822","branch":"HEAD"},"phetcommon":{"sha":"85d72626","branch":"HEAD"},"query-string-machine":{"sha":"b8bdcc28","branch":"HEAD"},"scenery":{"sha":"972ca6d7","branch":"HEAD"},"scenery-phet":{"sha":"3ee08581","branch":"HEAD"},"sherpa":{"sha":"9ca67479","branch":"HEAD"},"sun":{"sha":"720214f9","branch":"HEAD"},"tandem":{"sha":"6db63b4b","branch":"HEAD"}}

related to phetsims/QA/issues/84

**Comment by jessegreenberg** [comment]
Verified on my machine. It isn't great on the deployed version either, but seems to have gotten a bit worse since the last deploy.

Published version fluctuates between 15-30 fps in Edge
1.4.0-rc.1 goes fluctuates between 7-15 fps in Edge

**Comment by jessegreenberg** [comment]
@ariel-phet can you please prioritize this issue given the performance of the deployed version? Should we investigate?

**Comment by jessegreenberg** [comment]
I checked in with @phet-steele, he is not aware of Edge performance degradation for other sims.

**Comment by jessegreenberg** [comment]
Framerate looks better when particle movemennt animation removed (but still adding/removing from the view).  Did we change the clip area implementation in this sim?

**Comment by jessegreenberg** [comment]
Doesn't look like clip has changed, we are still using the clip workaround for Safari:

```js
// TODO: This clip area has been replaced with a layered rectangle in MoleculesAndLightScreenView because of a
// Safari specific SVG bug...
```

**Comment by jessegreenberg** [comment]
I will wait to spend more time on this until hearing whether or not it is a priority.

**Comment by ariel-phet** [comment]
@jessegreenberg yes worth investigating (even if you don't necessarily fix it). We will be working with MS to make a Win 10 version of our app, and making such an app one the windows platform will require us using a webview in Edge. So it would be good to be able to show MS some places where Edge is performing poorly.

**Comment by jessegreenberg** [comment]
Rendering photons with SVG brings framerate up to 60fps!

```js
    var photonLayer = new Node( { renderer: 'svg' } );
```
Performance is still bad when changing opacity of the photon emitter is still pretty bad. Maybe different renderer will help there too?

**Comment by jessegreenberg** [comment]
Same for the photon emitter, changing opacity of the photon emitter is very fluid when using SVG.

```js
    var photonEmitterLayer = new Node( { renderer: 'svg' } );
```

**Comment by jessegreenberg** [comment]
Committed above, pending testing.

**Comment by jessegreenberg** [comment]
Merged into 1.4 release branch.

**Comment by jessegreenberg** [comment]
@phet-steele pointed out that moving mouse over the radio button groups (and other things with hover styles for that matter) makes the performance drop. I am going to try setting sim renderer to svg and see if that helps.

**Comment by jessegreenberg** [comment]
Rendering the whole sim with svg in canvas seems to really improve performance. I don't see any weird rendering artifacts that have been reported in other sims. I am going to make a new RC with this change.

**Comment by jessegreenberg** [comment]
Merged into the 1.4 release branch.

**Comment by phet-steele** [comment]
Performance is quite spectacular now.
